packages/dds/tree/src/test/shared-tree/editing.spec.ts

```diff
@@ -2159,6 +2184,42 @@ describe("Editing", () => {
 			unsubscribe();
 		});
 
+		describe("Transactions", () => {
+			// Exercises a scenario where a transaction's inverse must be computed as part of a rebase sandwich.
+			it("Can rebase a series of edits including a transaction", () => {
+				const tree = makeTreeFromJson(["42"]);
+				const tree2 = tree.fork();
+
+				tree2.transaction.start();
+				tree2.editor.optionalField(rootField).set(singleJsonCursor("43"), false);
+				tree2.editor.optionalField(rootField).set(singleJsonCursor("44"), false);
+				tree2.transaction.commit();
+
+				tree2.editor.optionalField(rootField).set(singleJsonCursor("45"), false);
+
+				tree.editor.optionalField(rootField).set(singleJsonCursor("46"), false);
+
+				tree2.rebaseOnto(tree);
+				tree.merge(tree2, false);
+
+				expectJsonTree([tree, tree2], ["45"]);
+			});
+
+			it("Can set and delete a node within a transaction", () => {
+				const tree = makeTreeFromJson([]);
+				const tree2 = tree.fork();
+
+				tree2.transaction.start();
+				tree2.editor.optionalField(rootField).set(singleJsonCursor("42"), true);
+				tree2.editor.optionalField(rootField).set(undefined, false);
+				tree2.transaction.commit();
+
+				tree2.rebaseOnto(tree);
```

Comment:
`tree` has had no edits applied to it. If that's by design then this line can go away